### PR TITLE
Feature/running queries api endpoint and kill queries api endpoint

### PR DESCRIPTION
**Pull request by mkeagar**
Added AdminResource.java with API endpoints for:
- returning a list of the currently running queries on a KairosDB server along with the current count of queries waiting to run
- killing a currently running query by the query hash returned from the running queries endpoint.

### CODE REVIEW

File: src/main/java/org/kairosdb/core/datastore/QueryQueuingManager.java [hunk 1]
```diff
@@ -113,6 +114,39 @@ private boolean acquireSemaphore(String queryHash, QueryMetric metric) throws In
...
 
+	public ArrayList<Pair<String, QueryMetric>> getRunningQueries()
+	{
+		ArrayList<Pair<String, QueryMetric>> runningQueriesList = new ArrayList<Pair<String, QueryMetric>>();
+		lock.lock();
+		ArrayList<String> queryKeys = new ArrayList<String>();
+		queryKeys.addAll(runningQueries.keySet());
+		try
+		{
+			for (String key : queryKeys)
```
Comment (brianhks) [inline src/main/java/org/kairosdb/core/datastore/QueryQueuingManager.java]: Just put runningQueries.keySet() here in the for loop.  No need for the extra queryKeys array list that you copy it into.

File: src/main/java/org/kairosdb/core/datastore/QueryQueuingManager.java [hunk 2]
```diff
@@ -113,6 +114,39 @@ private boolean acquireSemaphore(String queryHash, QueryMetric metric) throws In
...
+		}
+		return runningQueriesList;
+	}
+
+	public void killQuery(String queryHash)
+	{
+		lock.lock();
+		try
+		{
+			runningQueries.get(queryHash).getSecond().interrupt();	// Call interrupt on Thread associated with provided query hash
```
Comment (brianhks) [inline src/main/java/org/kairosdb/core/datastore/QueryQueuingManager.java]: You need to check for null after the first get as the query may have finished before you call this to kill it.

File: src/main/java/org/kairosdb/core/datastore/QueryQueuingManager.java (hunk 1 above)
Comment (mkeagar) [reply src/main/java/org/kairosdb/core/datastore/QueryQueuingManager.java]: Yeah, I thought about doing that at first but decided against it... I've gone ahead and made this change now.

File: src/main/java/org/kairosdb/core/datastore/QueryQueuingManager.java (hunk 2 above)
Comment (mkeagar) [reply src/main/java/org/kairosdb/core/datastore/QueryQueuingManager.java]: @brianhks I've added an if statement to check if the runningQueries.get() results in a null, but should I add an else branch and throw an exception or return a message saying, "couldn't kill this query, it was already gone"? If the query is already gone, it seems to me that the desired result is already in effect.

Comment (brianhks) [reply src/main/java/org/kairosdb/core/datastore/QueryQueuingManager.java]: I wouldn't add an else.  Like you said if the query is gone then mission accomplished.

**Comment by kyleboyle** [comment]
Don't forget to update the api docs :)

**Comment by brianhks** [comment]
I was just thinking about that after I merged this, then saw your comment.

**Comment by brianhks** [comment]
Fixed issue #20

**Pull request merged**